src/rest/class-wp-rest-abilities-list-controller.php

```diff
@@ -0,0 +1,292 @@
+<?php declare( strict_types = 1 );
+
+/**
+ * REST API: WP_REST_Abilities_List_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 0.1.0
+ */
+
+/**
+ * Core controller used to access abilities via the REST API.
+ *
+ * @since 0.1.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Abilities_List_Controller extends WP_REST_Controller {
+
+	/**
+	 * REST API namespace.
+	 *
+	 * @since 0.1.0
+	 * @var string
+	 */
+	protected $namespace = 'wp/v2';
+
+	/**
+	 * REST API base route.
+	 *
+	 * @since 0.1.0
+	 * @var string
+	 */
+	protected $rest_base = 'abilities';
+
+	/**
+	 * Registers the routes for abilities.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[a-zA-Z0-9\-\/]+)',
+			array(
+				'args' => array(
+					'id' => array(
+						'description' => __( 'Unique identifier for the ability.' ),
+						'type'        => 'string',
+						'pattern'     => '^[a-zA-Z0-9\-\/]+$',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all abilities.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$abilities = wp_get_abilities();
```

Comment:
I see that it's a common practice in WP core to bail early for the `HEAD` method:

```php
if ( $request->is_method( 'HEAD' ) ) {
	// Return early as this handler doesn't add any response headers.
	return new WP_REST_Response( array() );
}
```

Reply:
I think this is a good idea. I added a TODO for it.

Reply:
Makes perfect sense.